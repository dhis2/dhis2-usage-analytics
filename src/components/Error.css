.uaa-error {
    padding: 32px 0;
    color: #e53935;
    text-align: center;
}
