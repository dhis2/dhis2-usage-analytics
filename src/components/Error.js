import React from 'react'
import PropTypes from '@dhis2/prop-types'
import './Error.css'

function Error({ message }) {
    return <div className="uaa-error">{message}</div>
}

Error.propTypes = {
    message: PropTypes.string,
}

export default Error
