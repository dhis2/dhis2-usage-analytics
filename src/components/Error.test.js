import React from 'react'
import { shallow } from 'enzyme'
import Error from './Error'

const testMessage = 'Test error message'
const defaultProps = {
    message: testMessage,
}

describe('<Error/>', () => {
    it('Matches the snapshot', () => {
        const tree = shallow(<Error {...defaultProps} />)
        expect(tree).toMatchSnapshot()
    })
    it('Renders renders the message passed to it', () => {
        const wrapper = shallow(<Error {...defaultProps} />)
        expect(wrapper.find('div.uaa-error').text()).toEqual(testMessage)
    })
})
