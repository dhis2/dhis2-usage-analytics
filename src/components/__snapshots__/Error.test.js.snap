// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> Matches the snapshot 1`] = `
<div
  className="uaa-error"
>
  Test error message
</div>
`;
